clang/test/Driver/unsupported-option-gpu.c

```diff
@@ -0,0 +1,5 @@
+/// Some target-specific options are ignored for GPU, so %clang exits with code 0.
+// DEFINE: %{check} = %clang -### -c -mcmodel=medium
```

Comment:
> Also, what exactly are we checking here? With `-###` CC1 sub-compilations do not run and, I think, that's where unsupported options were triggering warning/errors.

We're still only running the top-level driver. Weren't errors reported by cc1 compilation?

Reply:
Probably depends on the option we're testing. We could do both.

Reply:
In this particular case, the changes we test (and the error messages) were originating in the driver, so we're OK with `-###`.

That said, we do have other options that we do need to suppress/ignore and some of that does happen on cc1 level. 

We'll eventually need to add the tests for those here and then enable actual compilation, but it should not hold this patch back.

Reply:
In general the error reporting is done in the driver and cc1 allows and ignores options that may be specific to other targets.

> That said, we do have other options that we do need to suppress/ignore and some of that does happen on cc1 level.

Perhaps we should figure out them and check whether they should be moved to the driver